core/test/allocation.jl

```diff
@@ -29,47 +30,56 @@ end
 
 @testset "Simulation with allocation" begin
     toml_path =
-        normpath(@__DIR__, "../../generated_testmodels/simple_subnetwork/ribasim.toml")
+        normpath(@__DIR__, "../../generated_testmodels/minimal_subnetwork/ribasim.toml")
     @test ispath(toml_path)
     model = Ribasim.run(toml_path)
-    record = model.integrator.p.user.record
+    record = DataFrame(model.integrator.p.user.record)
     where_5 = (record.user_node_id .== 5)
-    where_6 = .!where_5
+    n_datapoints = sum(where_5)
 
-    @test all(record.demand[where_5] .== 1.0e-3)
+    record_5 = record[where_5, :]
+    record_6 = record[.!where_5, :]
+
+    @test all(record_5.demand .== 1.0e-3)
     @test all(
         isapprox(
-            record.allocated[where_5],
-            collect(range(1.0e-3, 0.0, sum(where_5)));
+            record_5.allocated,
+            collect(range(1.0e-3, 0.0, n_datapoints));
             rtol = 0.01,
         ),
     )
     @test all(
         isapprox(
-            record.abstracted[where_5][2:end],
-            collect(range(1.0e-3, 0.0, sum(where_5)))[2:end];
+            record_5.abstracted[2:end],
+            collect(range(1.0e-3, 0.0, n_datapoints))[2:end];
             rtol = 0.01,
         ),
     )
     @test all(
         isapprox(
-            record.demand[where_6],
-            collect(range(1.0e-3, 2.0e-3, sum(where_5)));
+            record_6.demand,
+            collect(range(1.0e-3, 2.0e-3, n_datapoints));
             rtol = 0.01,
         ),
     )
     @test all(
         isapprox(
-            record.allocated[where_6],
-            collect(range(1.0e-3, 2.0e-3, sum(where_5)));
+            record_6.allocated,
+            collect(range(1.0e-3, 2.0e-3, n_datapoints));
             rtol = 0.01,
         ),
     )
     @test all(
         isapprox(
-            record.abstracted[where_6][2:end],
-            collect(range(1.0e-3, 2.0e-3, sum(where_5)))[2:end];
+            record_6.abstracted[2:end],
+            collect(range(1.0e-3, 2.0e-3, n_datapoints))[2:end];
             rtol = 0.01,
         ),
     )
+
+    allocation_output_path = normpath(
+        @__DIR__,
+        "../../generated_testmodels/minimal_subnetwork/results/allocation.arrow",
+    )
+    @test ispath(allocation_output_path)
```

Comment:
```suggestion
    @test isfile(allocation_output_path)
```
`ispath` is for either files or folders, so this is a bit more specific.